server/jetstream_cluster.go

```diff
@@ -1315,8 +1315,8 @@ func (js *jetStream) monitorStream(mset *stream, sa *streamAssignment) {
 			if err := js.applyStreamEntries(mset, ce, isRecovering); err == nil {
 				n.Applied(ce.Index)
 				ne := ce.Index - lastApplied
-				// If over our compact min and we have at least min entries to compact, go ahead and snapshot/compact.
-				if _, b := n.Size(); b > compactSizeMin && ne >= compactNumMin {
+				// If we have at least min entries to compact, go ahead and snapshot/compact.
+				if ne >= compactNumMin {
```

Comment:
The `monitorConsumer()` has the same logic and compacts if size is above 4MB actually. Knowing that we saw cases where stream was compacted way too often (size was wrong metric and count was set to 4 instead of higher number), should we do the same there?